packages/formik/src/Formik.tsx

```diff
@@ -582,18 +582,20 @@ export function useFormik<Values extends FormikValues = FormikValues>({
   );
 
   const setFieldValue = useEventCallback(
-    (field: string, value: any, shouldValidate?: boolean) => {
+    (field: string, value: React.SetStateAction<any>, shouldValidate?: boolean) => {
+      const resolvedValue = isFunction(value) ? value(state.values[field]) : value;
```

Comment:
we can use `getIn` instead of `value(state.values[field])`.

E.g
```suggestion
const resolvedValue = isFunction(value) ? value(getIn(state.values, field)) : value;
```

Reply:
Fixed